egs/sre16/v2/get_egs_v2.sh

```diff
@@ -0,0 +1,298 @@
+  utils/filter_scp.pl $data/valid_set/utt2spk $data/utt2int > $data/valid_set/utt2int
+  utils/filter_scp.pl $data/train_subset/utt2spk $data/utt2int > $data/train_subset/utt2int
+  #Above, prepare the "utt2num_frames, utt2spk and utt2int" for each data set.
+fi
+
+num_pdfs=$(awk '{print $2}' $data/utt2int | sort | uniq -c | wc -l)
+num_train_frames=$(awk '{n += $2} END{print n}' <$data/train_set/utt2num_frames)
+num_train_subset_frames=$(awk '{n += $2} END{print n}' <$data/train_subset/utt2num_frames)
+#For here, the options are hardcode as we want to compare with the result of David
+echo $num_train_frames > $dir/info/num_frames
+num_train_archives=149
+echo $num_train_archives > $dir/info/num_archives
+
+if [ $nj -gt $num_train_archives ]; then
+  echo "$0: Reducing num-jobs $nj to number of training archives $num_train_archives"
+  nj=$num_train_archives
+fi
+#In David's setup, there are 149 archives. Each archives contains 70 million frames.
+#There are about 100 kinds of different length
+#So each "segment length" should have 70million frames * 149 / 100 = 104.3million
+
+if [ $stage -le 1 ]; then
+  echo "$0: Allocating training egs"
+  #Generate data_li/feat.scp and utt2spk
+  $cmd $dir/log/allocate_examples.log \
+    allocate_egs_seg_v2.py \
+      --min-frames-per-chunk=$min_frames_per_chunk \
+      --max-frames-per-chunk=$max_frames_per_chunk \
+      --kinds-of-length=$kinds_of_length_train \
+      --frames-per-kind=$frames_per_kind \
+      --data-dir=$data/train_set \
+      --output-dir=${data}/train_set/temp || exit 1
+  #sort and uniq
+  for subdir in `dir $data/train_set/temp`; do
+    cat $data/train_set/temp/$subdir/feats.scp.temp | sort | uniq > $data/train_set/temp/$subdir/feats.scp
+    cat $data/train_set/temp/$subdir/utt2spk.temp | sort | uniq > $data/train_set/temp/$subdir/utt2spk
+  done
+  #Generate data_li/spk2utt and utt2label
+  for subdir in `dir $data/train_set/temp`; do
+    utils/utt2spk_to_spk2utt.pl $data/train_set/temp/$subdir/utt2spk > $data/train_set/temp/$subdir/spk2utt
+    utils/sym2int.pl -f 2 $data/spk2int $data/train_set/temp/$subdir/utt2spk \
+      > $data/train_set/temp/$subdir/utt2int
+  done
+fi
+
+if [ $stage -le 2 ]; then
+  echo "$0: Combine, shuffle and split list for training set"
+  #combine feats.scp and shuffle
+  mkdir -p $data/train_set/temp/combine
+  data_combine=$data/train_set/temp/combine
+
+  cat $data/train_set/temp/data_*/feats.scp > ${data_combine}/feats.scp.bak
+  utils/shuffle_list.pl ${data_combine}/feats.scp.bak > ${data_combine}/feats.scp
+  #combine utt2spk
+  cat $data/train_set/temp/data_*/utt2spk > ${data_combine}/utt2spk
+  #generate spk2utt and utt2label
+  utils/utt2spk_to_spk2utt.pl ${data_combine}/utt2spk > ${data_combine}/spk2utt
+  utils/sym2int.pl -f 2 ${data}/spk2int ${data_combine}/utt2spk \
+    > ${data_combine}/utt2int
+
+  # split feats.scp
+  directories=$(for n in `seq $num_train_archives`; do echo ${data_combine}/split/${n}; done)
```

Comment:
It's not clear why you need to split the data directory. 
The get-egs binary requires only the feats.scp file. You can generate that by shuffling feats.scp to shuffled_feats.scp and using split_scp.pl. You don't need to even create the separate split directories. Just use something like shuffled_feats.$n.scp in the same directory, which you can delete at the end of the script.